internal/services/http.go

```diff
@@ -0,0 +1,64 @@
+package services
+
+import (
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+func GetGetRequestForUrl(url string) (*http.Request, error) {
+
```

Comment:
Whoops! Let's remove that empty line here. 

A better name for this function would be `PrepareRequestForURL` or something similar because `GetGet` sounds like a getter, which [should be avoided](https://go.dev/doc/effective_go#Getters) in Go.